controllers/storageclaim_controller.go

```diff
@@ -49,29 +50,54 @@ import (
 )
 
 const (
-	storageClassClaimFinalizer  = "storageclassclaim.ocs.openshift.io"
-	storageClassClaimAnnotation = "ocs.openshift.io/storagesclassclaim"
+	storageClaimFinalizer  = "storageclaim.ocs.openshift.io"
+	storageClaimAnnotation = "ocs.openshift.io/storageclaim"
+
+	pvClusterIDIndexName = "index:persistentVolumeClusterID"
+	vscDriverIndexName   = "index:volumeSnapshotContentCSIDriver"
 )
 
-// StorageClassClaimReconciler reconciles a StorageClassClaim object
-type StorageClassClaimReconciler struct {
+// StorageClaimReconciler reconciles a StorageClaim object
+type StorageClaimReconciler struct {
 	client.Client
 	cache.Cache
 	Scheme            *runtime.Scheme
 	OperatorNamespace string
 
-	log               logr.Logger
-	ctx               context.Context
-	storageClient     *v1alpha1.StorageClient
-	storageClassClaim *v1alpha1.StorageClassClaim
+	log           logr.Logger
+	ctx           context.Context
+	storageClient *v1alpha1.StorageClient
+	storageClaim  *v1alpha1.StorageClaim
 }
 
 // SetupWithManager sets up the controller with the Manager.
-func (r *StorageClassClaimReconciler) SetupWithManager(mgr ctrl.Manager) error {
+func (r *StorageClaimReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	ctx := context.Background()
+	if err := mgr.GetCache().IndexField(ctx, &corev1.PersistentVolume{}, pvClusterIDIndexName, func(o client.Object) []string {
+		csiSpec := o.(*corev1.PersistentVolume).Spec.CSI
+		if csiSpec != nil ||
+			csiSpec.Driver != csi.GetRBDDriverName() ||
+			csiSpec.Driver != csi.GetCephFSDriverName() ||
```

Comment:
Do we need these two checks? I am not sure is good for us to tie into the specific driver names. They are different downstream and upstream.

Reply:
in that scenario shouldn't the function return corresponding driver names as it is the source of truth at many places 🤔?

Reply:
So you want to index the provisioner name. I am ok with that approach